source/StellarObject.h

```diff
@@ -64,11 +74,22 @@ class StellarObject : public Body {
 	int Parent() const;
 	// Find out how far this object is from its parent.
 	double Distance() const;
+	// Defends the system using its defense weapons.
+	void TryToFire(std::vector<Projectile> &projectiles, const System *system, std::vector<Visual> &visuals, const std::list<std::shared_ptr<Ship>> &ships) const;
+	
+private:
 	
 	
 private:
 	const Planet *planet;
 	
+	const Weapon *launcher = nullptr;
+	//"overrides" Body.angle
+	mutable Angle angle;
+	
+	double reload = 1;
+	double rotation = 0;
```

Comment:
I would argue this should be named `turn`. Rotation has a special meaning when you are describing objections orbiting in space. Further, this rotation is only used when the orbiting object *is attempting to fire* - normal time passage would not see this object rotating unless the standard method - animated sprite - is used.

Reply:
where can i find that method?

Reply:
https://github.com/endless-sky/endless-sky/wiki/CreatingShips#sprites